redbot/cogs/downloader/repo_manager.py

```diff
@@ -1125,22 +1125,51 @@ def get_all_cogs(self) -> Tuple[Installable, ...]:
         old, new = await repo.update()
         return (repo, (old, new))
 
-    async def update_all_repos(self) -> Dict[Repo, Tuple[str, str]]:
-        """Call `Repo.update` on all repositories.
+    async def update_repos(
+        self, upd_repo_names: List[Repo] = []
```

Comment:
Mutable default arguments are dangerous in python.
```suggestion
        self, upd_repo_names: Optional[List[Repo]] = None
```

It doesn't look like any other handling for this should be necessary as the empty check for this below was using container truthfulness